### ISSUE TITLE
consider alternative names for `search_data` and `search_datasets`

### ISSUE THREAD

**Issue by itcarroll** [nsidc/earthaccess#770]
> Maybe we should think about the the top-level API; are these good names for these functions? Why not use common terminology like search_collections and search_granules? "Dataset" is often used to refer to a single data file. 🤔

~ @mfisher87 in #769 

Since "granules" is not very generic either, an option borrowed from the STAC spec could be "search_collections" vs "search_items".

Seems like a Milesone 1.0 change though ...

**Comment by chuckwondo** [comment]
I agree that we should consider alternative function names. My preference would be to use the prefix `find_` instead of `search_`, but not fussed about it.

**Comment by mfisher87** [comment]
I have no strong preference on search v find, but I am curious what reasoning underlies your preference @chuckwondo?

**Comment by chuckwondo** [comment]
> I have no strong preference on search v find, but I am curious what reasoning underlies your preference @chuckwondo?

Aside from it being 2 letters shorter, in previous contexts, I've often seen DB client APIs using `find` (or `find_by_X`) as a naming convention, so it is anecdotally arguably more consistent with other things. However, `search` may be equally widely used, so again, I don't have any overtly strong preference. It's just a mild preference, perhaps more personal than logical.

**Comment by mfisher87** [comment]
Thanks for expounding!

**Comment by betolink** [comment]
I like the idea of aligning with STAC, a while ago Scott suggested that and I think it'll be valuable to avoid cognitive load from users, the one thing I'm afraid is to deprecated existing names. I think we should try to not break the API to the extend possible while encouraging people to use the new conventions.  See: https://github.com/nsidc/earthaccess/discussions/221

**Comment by asteiker** [comment]
I know this has caused confusion in the past, even as others at NSIDC have come up to speed on the library, so I fully support updated names here! I like what @itcarroll proposed to align with STAC. We may also want to consider the language most commonly used within the NASA Earthdata ecosystem. You can't have Earthdata Search without "search", for example, so I'd be more keen on using this vs "find".  As an aside, this seems like a great use case for #761 too.

**Comment by chuckwondo** [comment]
Just to clarify, is this the current proposal?

- rename `search_datasets` to `search_collections`
- rename `search_data` to `search_items`

If so, +1 from me.

**Comment by mfisher87** [comment]
>As an aside, this seems like a great use case for https://github.com/nsidc/earthaccess/issues/761 too.

:rocket: :100:





> Just to clarify, is this the current proposal?
> 
>     * rename `search_datasets` to `search_collections`
> 
>     * rename `search_data` to `search_items`
> 
> 
> If so, +1 from me.

+1

>one thing I'm afraid is to deprecated existing names. I think we should try to not break the API to the extend possible while encouraging people to use the new conventions

I do worry that having multiple aliases for common features could lead to confusion, as people might think they do different things. I really like having "one correct way". I do believe a long deprecation period would be in order for top-level API things.

We need to probably have deeper discussions about how to communicate around time-until-deprecation. Should we always include a minimum date in our deprecation messages, e.g. `DeprecationWarning(" ... Obsoletion will occur no sooner than YYYY-MM-DD.")`?

**Comment by andypbarrett** [comment]
I like the alignment of `earthaccess` terminology with STAC.  _collections_ already aligns in STAC and NASA-speak.  However, as a newbie to STAC lingo, I find the usage of _items_ unclear.

**Comment by chuckwondo** [comment]
> > I have no strong preference on search v find, but I am curious what reasoning underlies your preference @chuckwondo?
> 
> Aside from it being 2 letters shorter, in previous contexts, I've often seen DB client APIs using `find` (or `find_by_X`) as a naming convention, so it is anecdotally arguably more consistent with other things. However, `search` may be equally widely used, so again, I don't have any overtly strong preference. It's just a mild preference, perhaps more personal than logical.

@mfisher87, after looking at the proposed new names again -- `search_collections` and `search_items` -- I now have an arguably better reason for preferring `find_collections` and `find_items`: The term `search_collections` is arguably ambiguous in terms of the types of "things" it will _find_. Does it search the available collections to find things _within_ collections, or does it search _for_ collections?

This analogy might be a bit of a stretch, but consider the case of security procedures at a place/event, where people may be subject to a "bag search." In that context, nobody is searching _for_ bags, they are searching _within_ bags (for banned "items"). Thus, the security folks running a `search_bags` function don't expect the result to be a "list of bags," but rather a list of "banned items" _within_ given bags.

Thus, I would argue that a "collection search" implemented by a function named `search_collections` could easily be misinterpreted to mean a search for items _within_ collections, not a search _for_ collections, or to simply cause someone to wonder which interpretation is correct, if they recognize the ambiguity. Thus, the name `find_collections` arguably eliminates such ambiguity by explicitly stating what we expect to find: _collections_. (Similarly for `find_items`.)

> I like the alignment of `earthaccess` terminology with STAC. _collections_ already aligns in STAC and NASA-speak. However, as a newbie to STAC lingo, I find the usage of _items_ unclear.

@andypbarrett, I agree that "items" is perhaps too generic for many folks. Although "collections" is perhaps no less generic a term, anecdotally, it may feel more specific to most folks dealing with this information. I don't have any particular preference or suggestion for a better term than "items," but if you have any suggestions, please share so we can "vote" on it here.

**Comment by mfisher87** [comment]
> Thus, I would argue that a "collection search" implemented by a function named `search_collections` could easily be misinterpreted to mean a search for items _within_ collections, not a search _for_ collections, or to simply cause someone to wonder which interpretation is correct, if they recognize the ambiguity. Thus, the name `find_collections` arguably eliminates such ambiguity by explicitly stating what we expect to find: _collections_. (Similarly for `find_items`.)

:100: This is an _excellent_ point. I'm on team _find_ now :)

**Comment by itcarroll** [comment]
Continuing discussion in #926.
